fix error if claim value equals bool false

alanpachuau opened the pull request:
Hasura expects claim value to be text. Without this fix, if the user_field value is false, the claim value is returned as boolean false. Hasura then triggers an error.

A check for undefined and null value of the user_field are already in place, so no need to check again in this line.

plmercereau commented:
I will take your bug fix into account in V2.
It's probably better to use `JSON.stringify` rather than `toString` as custom fields could be jsonb too, something like `JSON.stringify(value ?? null)`

elitan commented:
Could you guys provide me with an example where this would be an issue with the current code?

alanpachuau commented:
> Could you guys provide me with an example where this would be an issue with the current code?

In our application, we have added additional columns on the **users** table that we have added in our claims. One of the fields is boolean type. So, if the field value is false, the following line in auth-functions.js returns non-string value.

`custom_claims['x-hasura-' + user_field.replace('_', '-')] = user[user_field] && user[user_field].toString();`

This is the error message:
` x-hasura-* claims: expected Text, encountered Boolean`
